File: backend/src/main/java/com/primatoos/backend/model/Project.java
package com.primatoos.backend.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "projects")
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class Project {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String client;

    private String address;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "responsible_user_id", nullable = false)
    private User responsibleUser;

    @Column(name = "start_date")
    private LocalDate startDate;

    @Column(name = "expected_deadline")
    private LocalDate expectedDeadline;

    @Column(name = "current_stage")
    private String currentStage;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ProjectStatus status;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @Setter(AccessLevel.NONE)
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
    }
}

File: frontend/src/types/project.ts
import type { UserSummaryResponse } from './common'

export type ProjectStatus = 'PLANNING' | 'IN_PROGRESS' | 'PAUSED' | 'FINISHED' | 'CANCELLED'

export interface ProjectResponse {
  id: number
  name: string
  client: string
  address: string | null
  responsibleUser: UserSummaryResponse
  startDate: string | null
  expectedDeadline: string | null
  currentStage: string | null
  status: ProjectStatus
  notes: string | null
  createdAt: string
}

export interface ProjectSummaryResponse {
  id: number
  name: string
  client: string
}

export interface ProjectCreateRequest {
  name: string
  client: string
  address?: string | null
  responsibleUserId: number
  startDate?: string | null
  expectedDeadline?: string | null
  currentStage?: string | null
  notes?: string | null
}

export interface ProjectUpdateRequest {
  name: string
  client: string
  address?: string | null
  responsibleUserId: number
  startDate?: string | null
  expectedDeadline?: string | null
  currentStage?: string | null
  notes?: string | null
  status: ProjectStatus
}

export interface ProjectPhotoResponse {
  id: number
  url: string
  createdAt: string
}
